how-to/customize-workspace/client/src/connections.ts

```diff
@@ -0,0 +1,229 @@
+import { fin } from "@openfin/core";
+import { ChannelProvider } from "@openfin/core/src/api/interappbus/channel/provider";
+import { App } from "@openfin/workspace";
+import {
+	AppSourceConnection,
+	Connection,
+	ConnectionProviderOptions,
+	SnapshotSourceConnection
+} from "./connection-shapes";
+
+let connectionService: ChannelProvider;
+const connectedClients: { [key: string]: Connection } = {};
+const registeredActions: { [key: string]: () => Promise<void> } = {};
+let initialized = false;
+let connectionOptions: ConnectionProviderOptions;
+interface ConnectedAppSourceConnection {
+	identity: OpenFin.Identity;
+	connectionData: AppSourceConnection;
+}
+
+function isActionSupported(identity: OpenFin.Identity, payload: { action: string }) {
+	const requestingClient = connectedClients[identity.uuid];
+	const actions = "actions";
```

Comment:
Define `actions` in shapes as a const and use in other shape definitions as well

Reply:
the string actions is only used in the connectionProvider and nothing external uses it.